include/blmc_robots/real_finger.hpp

```diff
@@ -115,105 +118,218 @@ class RealFinger : public robot_interfaces::Finger {
+        // position where the index was last seen.
+        Vector encoder_index_angles = joint_modules_.get_measured_index_angles();
+        if (encoder_index_angles.hasNaN()) {
+            // Make sure we do not try to home on the index when the index
+            // was not found.  This should normally not happen, so no specific
+            // error handling is done here.
+            rt_printf("Homing failed, index not found.  Exit.\n");
+            exit(-1);
+        }
+
+        joint_modules_.set_zero_angles(encoder_index_angles + home_offset);
+
+        rt_printf("Home position: %f, %f, %f\n", encoder_index_angles[0],
+                  encoder_index_angles[1], encoder_index_angles[2]);
     }
 
+
+    /**
+     * @brief Move to given goal position using PD control.
+     *
+     * @param goal_pos Angular goal position for each joint.
+     * @param kp Gain K_p for the PD controller.
+     * @param kd Gain K_d for the PD controller.
+     * @param timeout_cycles Timeout.  If exceeded before goal is reached, the
+     *     procedure is aborted. Unit: Number of control loop cycles.
+     * @return True if goal position is reached, false if timeout is exceeded.
+     */
+    bool move_to_position(const Vector &goal_pos, const double kp,
+                          const double kd, const uint32_t timeout_cycles)
     {
+        /// \todo: this relies on the safety check in the motor right now,
+        /// which is maybe not the greatest idea. Without the velocity and
+        /// torque limitation in the motor this would be very unsafe
+
+        bool reached_goal = false;
+        int cycle_count = 0;
+        Vector desired_torque = Vector::Zero();
+        Eigen::Vector3d last_diff(std::numeric_limits<double>::max(),
+                                  std::numeric_limits<double>::max(),
+                                  std::numeric_limits<double>::max());
+
+        while (!reached_goal && cycle_count < timeout_cycles)
+        {
+            const TimeIndex t = append_desired_action(desired_torque);
+
+            const Vector position_error = goal_pos - get_observation(t).angle;
+            const Vector velocity = get_observation(t).velocity;
+
+            // we implement here a small PD control at the current level
+            desired_torque = kp * position_error - kd * velocity;
+
+#ifdef VERBOSE
+            if (cycle_count % 100 == 0) {
+                rt_printf("--\n"
+                          "position error: %10.5f, %10.5f, %10.5f\n"
+                          "velocity:       %10.5f, %10.5f, %10.5f\n",
+                          position_error[0], position_error[1], position_error[2],
+                          velocity[0], velocity[1], velocity[2]);
+            }
+#endif
+
+            // Check if the goal is reached (position error below tolerance and
+            // velocity close to zero).
+            constexpr double POSITION_TOLERANCE = 0.2;
```

Comment:
could we make this also a function parameter?

Reply:
done

Reply:
thanks!